### DISCUSSION TITLE
Issue with MVBS calculation for AZFP, missing ping_time dimension

### DISCUSSION THREAD

**Discussion by slonimer** [OSOceanAcoustics/echopype#440] [Q&A]
I am attempting to use echoPype for processing AZFP data.  I have extract the data to Sv values, using ep.calibrate.compute_Sv(echodata).  However, when I try to calculate MVBS (with ep.preprocess.compute_MVBS or ep.preprocess.compute_MVBS_index_binning), I get the error:
ValueError: Dimensions {'ping_time'} do not exist. Expected one or more of ('frequency', 'range_bin')

I have tried the example for the EK60 data and I am able to calculate MVBS without issue.  I have traced the problem to the fact that for the echodata variable, "range", the dimensions are: (frequency, range_bin), whereas for the EK60 data, the dimensions are: 
(frequency, ping_time, range_bin).  

I have tried to fix this by finding where these dimensions are set for the AZFP, so that I can add the missing ping_time dimension,   but I cannot figure this out!  Any help would be very much appreciated!

![EK60_dataset](https://user-images.githubusercontent.com/48665977/133682759-0b724932-45a4-475f-a231-907dff6261ce.png)
![AZFP_dataset](https://user-images.githubusercontent.com/48665977/133682764-3436e29c-7ad3-48c6-9887-b70a2bd5102b.png)

**Comment by slonimer** [comment]
This doesn't fix the core issue, but I found a work-around where I can add a dimension to the range variable, and then proceed with the MVBS calculations: 

ds_Sv["range"] = ds_Sv["range"].expand_dims(axis=1, ping_time = ds_Sv["ping_time"])

**Comment by leewujung** [comment]
@slonimer Thanks for reporting this! This is unexpected, since an earlier version retained the `ping_time` dimension. It is possibly due to some reorganization of the calibration functions, since for AZFP a subset of the env params have to be specified by the user, which could come in _without_ any dimensions attached. We are looking into this now!

**Reply by leewujung** [reply]
FYI -- brief investigation led to documenting the two issues found related to this #453 #454. We'll get this fixed soon.
